### ISSUE TITLE
Open print preview on init

### ISSUE THREAD

**Issue by i7i5** [stephanrauh/ngx-extended-pdf-viewer#640]
Hello, with the migration to angluar 10+ we switched from another inactive pdf lib to this one. The old one had an input field to open the print preview with opening the pdf (the parameter is called "@Input() public startPrint: boolean;". What do you think about such an implementation as we got questioned by our customers about the changed behavior?

**Describe the solution you'd like**
I'd like to have a flag to trigger the opening of the print preview on pdf viewer init. Like:
```ts
<ngx-extended-pdf-viewer
     [startPrint/openPrintPreviewOnInit smth]=true
>
</ngx-extended-pdf-viewer>
```

BR
i7i5

**Comment by stephanrauh** [comment]
Sounds like you've been using ng2-pdfjs-pdf-viewer. I can assure you that library isn't dead. Every once in a while, I exchange e-mails with the author. But you're right, for some reason unknown there's little activity over there. AFAIK the reason is trying to support Internet Explorer... so it should surprise nobody it takes a while. :)

**Comment by i7i5** [comment]
Yes wasn't sure if it is ok to mention it here. The author doesn't release new versions as there is an issue open to finally release the version that supports angular 10. Also he doesn't respond either to questions, thats why I thought its dead now :/

**Comment by stephanrauh** [comment]
As for your question: I'd prefer not to add a new attribute. Instead, I suggest using the event `(pdfLoaded)` to trigger printing using the print API. I haven't tried it, but I'm pretty sure this approach works out of the box. If it doesn't, you need an event that's triggered later, or you need to add a timeout. So the solution looks something like this:

```html
<ngx-extended-pdf-viewer
      [src]="'/assets/pdfs/Portugues-para-principiantes-1538054164.pdf'"
      [useBrowserLocale]="true"
      (pdfLoaded)="print()">
</ngx-extended-pdf-viewer>
```


```TypeScript
@Component({ ... })
export class PrintRangeComponent {
  constructor(private printService: NgxExtendedPdfViewerService) {}

  public print(): void {
    this.printService.print();
  }
}
```

**Comment by stephanrauh** [comment]
I'd appreciate your feedback, especially if the solution works for you, because that's when most people forget to share their joy with me. :)

Are you fine with me not adding the attribute? Should I add a demo to the showcase?

**Comment by i7i5** [comment]
Thanks a lot for the solution. I tried it in my test repo and pushed it to the following prompt: https://github.com/i7i5/pdf-example/tree/open-print-preview-on-init an get the following issue:

Console shows the following error: "Expected print service to be initialized." and a popup appears: "Warning: The PDF is not fully loaded for printing.". This is what happens:
When I click the button the viewer shows up but pdf isn't loaded yet. Then the prompt shows up and the event got fired (no sure whats first but i guess the event is before the prompt). After closing it the pdf got loaded correctly but no preview was opened.

Maybe this is a bug?

BR

**Comment by stephanrauh** [comment]
It's a timing problem. I've given you the wrong event. Use `(pagesLoaded)` instead. This even works with https://www.obwb.ca/library/okanagan-basin-waterscape-poster/, a file that takes more than a minute to render.

**Comment by stephanrauh** [comment]
![image](https://user-images.githubusercontent.com/3045767/107151464-fef34780-6962-11eb-8be3-3ef450b9187c.png)

**Comment by i7i5** [comment]
Nice thank you a lot again, I'll implement this solution tomorrow as bugfix :)
Have a nice day!

**Comment by stephanrauh** [comment]
@i7i5 I just saw I've confused ng2-pdf-viewer with ng2-pdfjs-viewer yesterday. These names are just to similar! I'm afraid you're right: ng2-pdfjs-viewer seems to be dead. That's a pity, because it's using a simple and rock-solid approach. The library I consider still alive - just a bit late - is https://www.npmjs.com/package/ng2-pdf-viewer. I've contributed a few bits to that library recently, so I really hope it's alive. :)
